unstructured_ingest/v2/processes/connectors/onedrive.py

```diff
@@ -100,6 +101,27 @@ class OnedriveIndexerConfig(IndexerConfig):
     recursive: bool = False
 
 
+T = TypeVar("T")
+
+
+def async_iterable_to_sync_iterable(iterator: AsyncIterator[T]) -> Iterator[T]:
+    # This version works on Python 3.9 by manually handling the async iteration.
+    loop = asyncio.new_event_loop()
```

Comment:
Hmm why do we need to create a new loop? :O can't we reuse the default one with `asyncio.get_event_loop()`?

Reply:
Ok, makes sense.